cmd/tegola/cmd/root.go

```diff
@@ -89,23 +60,31 @@ func rootCmdValidatePersistent(cmd *cobra.Command, _ []string) (err error) {
 		build.Commands = append(build.Commands, cmdName)
 		return nil
 	default:
-		return initConfig(configFile, requireCache, logLevel, logger)
+		return initConfig(configFile, requireCache, logLevel)
 	}
 }
 
-func initConfig(configFile string, cacheRequired bool, logLevel string, logger string) (err error) {
-	err = validateSupportedLoggers(logger)
-	if err != nil {
-		return err
-	}
-	log.SetLogger(logger)
-
-	// set log level before the first log is called
-	level, err := getLogLevelFromString(logLevel)
-	if err != nil {
-		return err
-	}
-	log.SetLogLevel(level)
+func initConfig(configFile string, cacheRequired bool, logLevel string) (err error) {
+	// Parse the provided log level; default to INFO if parsing fails.
+	lvl := log.ParseLogLevel(logLevel)
+
+	// Create a base text handler that outputs to stderr.
+	// The AddSource option includes file and line info in each log record.
+	baseHandler := slog.NewJSONHandler(os.Stderr, &slog.HandlerOptions{
+		Level: lvl,
+		// TODO: enable once we switch to slog.Default
+		// instead of internal/log methods
+		AddSource: false,
+	})
+
+	// Wrap the base handler with our custom handler to add stack traces for errors.
+	handler := log.NewHandler(baseHandler)
+	logger := slog.New(handler).
+		WithGroup("tegola").
+		With("version", build.Version, "pid", os.Getpid())
```

Comment:
we could add in the git hash and branch here as well: 

https://github.com/go-spatial/tegola/blob/9bac264ffa8d15b80423584c2480b0a811351e27/internal/build/build.go#L11-L13

Reply:
Ill add the revision, but not the branch. That's likely to be master anyways unless you're on a fork or test in a pr.